content/blog/jetstream-java-client-03-consume.md

```diff
@@ -0,0 +1,131 @@
+| Last | The consumer will start receiving messages with the last message. /
+| New | The consumer will only start receiving messages that were created after the consumer was created. /
+| ByStartSequence | The consumer is required to specify the `.startSequence(...)`, the sequence number to start on. It will receive the closest available sequence if that message was removed based on the stream limit policy. | 
+| ByStartTime | The consumer is required to specify the `.startTime(...)`, the time in the stream to start at. It will receive the closest available message on or after that time. | 
+
+### Deliver Subject
+
+Deliver subject essentially creates an alias core NATS subject for the stream. This means a core NATS subscriber could be set up to receive messages on the deliver subject,
+starting wherever the deliver policy was configured for the consumer. You could use a core NATS subscriber to access the stream by its original subject, but that would always start at
+the next message that appears on the subject, instead of where the consumer configured start sequence or start time. This is only allowed for push subscriptions.
+
+### Ack Policy
+
+`Explicit` is the default policy and means that each individual message must be acknowledged.
+For pull consumers, `Explicit` is the only allowed option.
+For push consumers, you can choose `None`, which means you do not have to ack any messages,
+or `All` which means whenever you choose to ack a message, all the previous messages received are automatically acknowledged.
+You must ack within the Ack Wait window or it will be as if the ack was not received.
+
+### Ack Wait
+
+Ack Wait is the time in nanoseconds that the server will wait for an ack for any individual message.
+If an ack is not received in time, the message will be redelivered.
+
+### Max Ack Pending
+
+If you have received this number of messages without acking them, the server will stop sending messages.
+If the server gets 1 or more acks, it will resume sending that number of messages.
+If acks don't occur in the Ack Wait period, then the server will resume starting at the messages that have not been acked.
+
+### Replay Policy
+
+The replay policy applies when the deliver policy is `All`, `ByStartSequence` or `ByStartTime` since those deliver policies begin reading the stream at a position other than the end.
+If the policy is `ReplayOriginal` (the default), the messages in the stream will be pushed to the client at the same rate that they were originally received, simulating the original timing of messages.
+If the policy is `ReplayInstant`, the messages will be pushed to the client as fast as possible while adhering to the Ack Policy, Max Ack Pending and the client's ability to consume those messages.
+
+### Max Deliver
+
+The maximum amount times a specific message will be delivered. Applies to messages that are re-sent due to ack policy.
```

Comment:
Think this could be a bit clearer

Reply:
copied from https://docs.nats.io/jetstream/concepts/consumers